packages/components/src/internal/components/picklist/models.ts

```diff
@@ -0,0 +1,42 @@
+import { Draft, immerable, produce } from 'immer';
+import { User } from '../base/models/User';
+import { PUBLIC_PICKLIST_CATEGORY } from '../domainproperties/list/constants';
+import { userCanDeletePublicPicklists, userCanManagePicklists } from '../../app/utils';
+
+export class PicklistModel {
+    [immerable] = true;
+
+    readonly Category: string;
+    readonly CreatedBy: number;
+    readonly name: string;
+    readonly listId: number;
+    readonly Description: string;
+
+    constructor(values?: Partial<PicklistModel>) {
+        Object.assign(this, values);
+    }
+
+    isUserList(user: User): boolean {
+        return this.CreatedBy === user.id
+    }
+
+    isEditable(user: User): boolean {
+        return this.isUserList(user) && userCanManagePicklists(user);
+    }
+
+    isPublic(): boolean {
+        return this.Category === PUBLIC_PICKLIST_CATEGORY;
+    }
+
+    isDeletable(user: User): boolean {
+        return (this.isUserList(user) && userCanManagePicklists(user)) || (this.isPublic() && userCanDeletePublicPicklists(user));
```

Comment:
the first part fo this OR statement looks to match the `isEditable`, can you just use that here?